text/0000-prepare-operation.md

```diff
@@ -0,0 +1,387 @@
+    - We've been wanting to restructure our spec repo to include other schemas, in JSON format, as well. Maybe this is an opportunity to rethink our repo structure.
+
+[lifecycle-config-rfc]: https://github.com/buildpacks/rfcs/pull/128
+
+# Spec. Changes
+[spec-changes]: #spec-changes
+
+## Distribution Spec
+
+No changes necessary.
+
+## Platform Spec
+
+### Add Project Descriptor
+
+---
+
+#### `project.toml` (TOML)
+
+The format of `project.toml` MUST adhere to version `0.2` of the [project descriptor specification][project-descriptor-spec]. Within the `project.toml` file the `io.buildpacks` namespace MAY be defined.
+
+[project-descriptor-spec]: https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md
+
+---
+
+### Add Prepare operation
+
+---
+
+#### Prepare
+
+Before the [Build](#Build) operation is executed, a platform MUST prepare the build environment.
+
+During, the Prepare phase, the platform:
+
+  - SHOULD apply provided [Project Descriptor][project-descriptor] configuration is present.
+      - SHOULD generate a warning for any `io.buildpacks` property not applied.
+  - MAY make changes to any path in the filesystem.
+
+[project-descriptor]: https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md
+
+---
+
+
+### Add `prepare` section
+
+---
+
+#### `preparer`
+
+Usage:
+```
+/cnb/lifecycle/preparer \
+  [-app <app>] \
+  [-buildpacks <buildpacks>] \
+  [-cache-dir <cache-dir>] \
+  [-cache-image <cache-image>] \
+  [-daemon] \
+  [-gid <gid>] \
+  [-launch-cache <launch-cache> ] \
+  [-launcher <launcher> ] \
+  [-layers <layers>] \
+  [-log-level <log-level>] \
+  [-order <order>] \
+  [-platform <platform>] \
+  [-previous-image <previous-image> ] \
+  [-process-type <process-type> ] \
+  [-project-descriptor <project-descriptor> ] \
+  [-project-metadata <project-metadata> ] \
+  [-report <report> ] \
+  [-run-image <run-image>] \
+  [-skip-restore <skip-restore>] \
+  [-stack <stack>] \
+  [-tag <tag>...] \
+  [-uid <uid> ] \
+  <image>
+```
+
+The `preparer` SHOULD accept the same inputs as the `creator` with the addition of the following:
+
+| Input             | Environment Variable| Default Value| Description
+|-------------------|---------------------|--------------|----------------------
+| `<project-descriptor>`| `CNB_PROJECT_DESCRIPTOR_PATH`| `<app>/project.toml`    | Path to a [Project Descriptor](#projecttoml-TOML)
+
+##### Outputs
+
+A `preparer` may make general changes to the file system, modify input files, or create input files.
+
+| Exit Code       | Result
+|---              |---
+| `0`             | Success
+| `1-10`, `13-19` | Generic lifecycle errors
+
+<!-- </details> -->
+
+---
+
+## Project Descriptor Spec
+
+#### Remove `io.buildpacks` namespace
+
+We'll want to remove the `io.buildpacks` namespace since it will be versioned seperate from the project descriptor.
```

Comment:
I guess I understand the need for this but I also find the idea of another spec somewhat confusing. Unfortunately I don't have any constructive suggestions to offer.